pyomo/core/expr/numeric_expr.py

```diff
@@ -557,10 +557,14 @@ def create_node_with_local_data(self, args, classtype=None):
         if npv_args:
             return super().create_node_with_local_data(args, None)
         else:
-            cls = list(self.__class__.__bases__)
-            cls.remove(NPV_Mixin)
-            assert len(cls) == 1
-            return super().create_node_with_local_data(args, cls[0])
+            return super().create_node_with_local_data(
+                args, self.potentially_variable_base_class())
+
+    def potentially_variable_base_class(self):
```

Comment:
Should this be a public method?

Reply:
I was torn.  I need the logic in two places, and one is not in this module (or in a derived class), so technically, I think the answer is "yes".  BUT I am not thrilled with it (nor the name).